pkg/cli/image_apk.go

```diff
@@ -0,0 +1,245 @@
+	cmd.Flags().StringSliceVarP(&p.distroDirPaths, "distro-dir", "d", nil, "path to a directory containing Melange build configuration files")
+}
+
+type syftResults struct {
+	sbom                                *sbom.SBOM
+	packageOwnershipsByOwnedPackageName map[string][]ownership
+}
+
+type ownership struct {
+	owner pkg.Package
+	owned pkg.Package
+}
+
+func (r *syftResults) apks() []pkg.Package {
+	var apks []pkg.Package
+
+	pkgs := r.sbom.Artifacts.Packages.Sorted(pkg.ApkPkg)
+	for i := range pkgs {
+		p := pkgs[i]
+		// Skip findings from SBOMs
+		if p.FoundBy == "sbom-cataloger" {
+			continue
+		}
+
+		apks = append(apks, p)
+	}
+
+	return apks
+}
+
+func (r *syftResults) apksOwningPackageWithName(name string) ([]pkg.Package, error) {
+	// Lazily create the index of owned package (by name) to APK package
+	if r.packageOwnershipsByOwnedPackageName == nil {
+		err := r.indexPackageOwnerships()
+		if err != nil {
+			return nil, fmt.Errorf("unable to index APK package ownerships: %w", err)
+		}
+	}
+
+	// Lookup the APK package that owns the given package name
+	ownerships, ok := r.packageOwnershipsByOwnedPackageName[name]
+	if !ok || len(ownerships) == 0 {
+		return nil, fmt.Errorf("no APK package owns %q", name)
+	}
+
+	var apks []pkg.Package
+	for i := range ownerships {
+		o := ownerships[i]
+		apks = append(apks, o.owner)
+	}
+
+	return apks, nil
+}
+
+func (r *syftResults) indexPackageOwnerships() error {
+	index := make(map[string][]ownership)
+
+	for _, rel := range r.sbom.Relationships {
+		if rel.Type != artifact.OwnershipByFileOverlapRelationship {
+			continue
+		}
+
+		apkPkgID := rel.From.ID()
+		ownedPkgID := rel.To.ID()
+
+		apkPkg := r.sbom.Artifacts.Packages.Package(apkPkgID)
+		if apkPkg == nil {
+			return fmt.Errorf("unable to find owner package %q", apkPkgID)
+		}
+		if apkPkg.Type != pkg.ApkPkg {
+			return fmt.Errorf(
+				"expected APK package as owner in relatioship, got %s (for package %q)",
+				apkPkg.Type,
+				apkPkg.Name,
+			)
+		}
+
+		ownedPkg := r.sbom.Artifacts.Packages.Package(ownedPkgID)
+		if ownedPkg == nil {
+			return fmt.Errorf("unable to find owned package %q", ownedPkgID)
+		}
+
+		o := ownership{
+			owner: *apkPkg,
+			owned: *ownedPkg,
+		}
+
+		index[ownedPkg.Name] = append(index[ownedPkg.Name], o)
+	}
+
+	r.packageOwnershipsByOwnedPackageName = index
+	return nil
+}
+
+func apkConfigPaths(apks []pkg.Package, distroDirPaths []string) ([]string, error) {
+	configIndexes := make([]*configs.Index[config.Configuration], 0, len(distroDirPaths))
+	indexRootPaths := make([]string, 0, len(distroDirPaths))
+
+	for _, p := range distroDirPaths {
+		index, err := build.NewIndex(rwos.DirFS(p))
+		if err != nil {
+			return nil, fmt.Errorf("unable to index configuration files from %q: %w", p, err)
+		}
+		configIndexes = append(configIndexes, index)
+		indexRootPaths = append(indexRootPaths, p)
+	}
+
+	configPaths := make([]string, 0, len(apks))
+
+	for i := range apks {
+		apk := apks[i]
```

Comment:
nit: 
```suggestion
	for i, apk := range apks {
```

Reply:
Switched back to that and added a `nolint` — also happy to strike the lint rule globally if we want!